### ISSUE TITLE
ERROR: Couldn't sync keys

### ISSUE THREAD

**Issue by basedtho** [Foxboron/sbctl#19]
```
 ~ sudo sbctl enroll-keys
==> Syncing /usr/share/secureboot/keys to EFI variables...
Invalid key /usr/share/secureboot/keys/PK/PK.der.esl
 - unknown cert type
Invalid key /usr/share/secureboot/keys/PK/PK.pem
 - unknown cert type
Invalid key /usr/share/secureboot/keys/PK/PK.der
 - unknown cert type
Invalid key /usr/share/secureboot/keys/PK/PK.key
 - unknown cert type
Invalid key /usr/share/secureboot/keys/KEK/KEK.der.esl
 - unknown cert type
Invalid key /usr/share/secureboot/keys/KEK/KEK.pem
 - unknown cert type
Invalid key /usr/share/secureboot/keys/KEK/KEK.der
 - unknown cert type
Invalid key /usr/share/secureboot/keys/KEK/KEK.key
 - unknown cert type
Invalid key /usr/share/secureboot/keys/db/db.der.esl
 - unknown cert type
Invalid key /usr/share/secureboot/keys/db/db.pem
 - unknown cert type
Invalid key /usr/share/secureboot/keys/db/db.der
 - unknown cert type
Invalid key /usr/share/secureboot/keys/db/db.key
 - unknown cert type
Filesystem keystore:
  /usr/share/secureboot/keys/db/db.auth [3285 bytes]
  /usr/share/secureboot/keys/db/db.der.esl [1263 bytes]
  /usr/share/secureboot/keys/db/db.pem [1708 bytes]
  /usr/share/secureboot/keys/db/db.der [1219 bytes]
  /usr/share/secureboot/keys/db/db.key [3272 bytes]
  /usr/share/secureboot/keys/KEK/KEK.auth [3281 bytes]
  /usr/share/secureboot/keys/KEK/KEK.der.esl [1271 bytes]
  /usr/share/secureboot/keys/KEK/KEK.pem [1716 bytes]
  /usr/share/secureboot/keys/KEK/KEK.der [1227 bytes]
  /usr/share/secureboot/keys/KEK/KEK.key [3272 bytes]
  /usr/share/secureboot/keys/PK/PK.auth [3273 bytes]
  /usr/share/secureboot/keys/PK/PK.der.esl [1263 bytes]
  /usr/share/secureboot/keys/PK/PK.pem [1708 bytes]
  /usr/share/secureboot/keys/PK/PK.der [1219 bytes]
  /usr/share/secureboot/keys/PK/PK.key [3272 bytes]
firmware keys:
  PK:
  KEK:
    /C=US/ST=Washington/L=Redmond/O=Microsoft Corporation/CN=Microsoft Corporation KEK CA 2011
    /C=JP/ST=Kanagawa/L=Yokohama/O=Lenovo Ltd./CN=Lenovo Ltd. KEK CA 2012
  db:
    /C=US/ST=Washington/L=Redmond/O=Microsoft Corporation/CN=Microsoft Windows Production PCA 2011
    /C=US/ST=Washington/L=Redmond/O=Microsoft Corporation/CN=Microsoft Corporation UEFI CA 2011
    /C=US/ST=North Carolina/O=Lenovo/CN=Lenovo UEFI CA 2014
    /C=JP/ST=Kanagawa/L=Yokohama/O=Lenovo Ltd./CN=ThinkPad Product CA 2012
  dbx:
    10d45fcba396aef3153ee8f6ecae58afe8476a280a2026fc71f6217dcf49ba2f
    <...>
    895a9785f617ca1d7ed44fc1a1470b71f3f1223862d9ff9dcc3ae2df92163daf
    82db3bceb4f60843ce9d97c3d187cd9b594Can't create key file /sys/firmware/efi/efivars/db-d719b2cb-3d3a-4596-a3bc-dad00e67656f: Operation not permitted
Error syncing keystore file /usr/share/secureboot/keys/db/db.auth
Can't create key file /sys/firmware/efi/efivars/KEK-8be4df61-93ca-11d2-aa0d-00e098032b8c: Operation not permitted
Error syncing keystore file /usr/share/secureboot/keys/KEK/KEK.auth
1cd3de8100e586f2bda5637575f67
    81a8b965bb84d3876b9429a95481cc955318cfaa1412d808c8a33bfd33fff0e4
    <...>
    80b4d96931bf0d02fd91a61e19d14f1da452e66db2408ca8604d411f92659f0a
filesystem keys:
  PK:
    /C=Platform Key
     from /usr/share/secureboot/keys/PK/PK.auth
  KEK:
    /C=Key Exchange Key
     from /usr/share/secureboot/keys/KEK/KEK.auth
  db:
    /C=Database Key
     from /usr/share/secureboot/keys/db/db.auth
  dbx:
New keys in filesystem:
 /usr/share/secureboot/keys/db/db.auth
 /usr/share/secureboot/keys/KEK/KEK.auth
 /usr/share/secureboot/keys/PK/PK.auth
Inserting key update /usr/share/secureboot/keys/db/db.auth into db
Inserting key update /usr/share/secureboot/keys/KEK/KEK.auth into KEK

==> ERROR: Couldn't sync keys
```

**Comment by Foxboron** [comment]
Can you clear the current secure boot keys in the firmware and try again?

**Comment by basedtho** [comment]
It worked! But I can’t boot anymore, because there’s no entry in the boot menu. What do I do?

**Comment by basedtho** [comment]
Okay, so what I did was - boot the livecd, `bootctl --esp-path=/boot install`, disable secure boot, finally boot into arch (not livecd), then do the rest of the instructions in the readme (signatures, generate efi stub), enable secure boot again, and then it won't let me boot, says `/EFI/Linux/linux-linux.efi not trusted` or something

**Comment by basedtho** [comment]
Okay, I've figured it out. I had to sign `/boot/EFI/Linux/linux-linux.efi` too, I did not at first because it was created after the `Signatures` step. Now it works!

**Comment by Foxboron** [comment]
Awesome :) Is there any instructions that should be made clearer? I have contemplated making a webpage with better documentation for `sbctl` and `goefi`.

**Comment by basedtho** [comment]
Dunno, tbh. But one thing for certain is that all $'s should be substituted with #'s, because It doesn't work without root

**Comment by basedtho** [comment]
Oh, btw, I assume it's not safe to store the keys in `/usr/share/secureboot` on an unencrypted partition? I use `systemd-homed`, and only my home folder is encrypted. Where do I put the keys?

**Comment by Foxboron** [comment]
> Dunno, tbh. But one thing for certain is that all $'s should be substituted with #'s, because It doesn't work without root

Good point.

>Oh, btw, I assume it's not safe to store the keys in /usr/share/secureboot on an unencrypted partition? I use systemd-homed, and only my home folder is encrypted. Where do I put the keys?

You are correct. There isn't support for secure storage for the keys yet. I want to try have some support for yubikeys, and potentially some key encryption.

I want to write support for a config file, but haven't gotten that far yet. It would allow you to have other storage locations.

**Comment by basedtho** [comment]
hey Morten! i know i'm supposed to create a new issue, but i'm sorry i just don't know where to begin
please take a look at https://www.reddit.com/r/archlinux/comments/hx7tar/failed_to_boot_after_upgrading_to_5710/

tldr: kernel can't find modules after upgrade to 5.7.10, it's still looking at the modules/5.7.9 directory, and even when i downgraded to 5.7.8, it still looked at modules/5.7.9. 

i think this has to do with sbctl

**Comment by Foxboron** [comment]
Reboot your computer. Look at the output from `uname -r` and `pacman -Q`.

**Comment by basedtho** [comment]
this is very weird

![image](https://user-images.githubusercontent.com/60894565/88434309-73173500-cdef-11ea-9f4c-ed76618ffe15.png)

**Comment by Foxboron** [comment]
Did you reboot?
